cfn-resources/org-invitation/cmd/resource/resource.go

```diff
@@ -0,0 +1,241 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	log "github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	mongodbatlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+var CreateRequiredFields = []string{constants.OrgID, constants.PubKey, constants.PvtKey, constants.Username}
+var ReadRequiredFields = []string{constants.OrgID, constants.ID, constants.PubKey, constants.PvtKey}
+var UpdateRequiredFields = []string{constants.OrgID, constants.ID, constants.PubKey, constants.PvtKey}
+var DeleteRequiredFields = []string{constants.OrgID, constants.ID, constants.PubKey, constants.PvtKey}
+var ListRequiredFields = []string{constants.PubKey, constants.PubKey}
+
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-OrgInvitation")
+}
+
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("Create() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(CreateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Warnf("Create - error: %+v", err)
+		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	invitationReq := &mongodbatlas.Invitation{
+		TeamIDs:  currentModel.TeamIds,
+		Roles:    currentModel.Roles,
+		Username: *currentModel.Username,
+	}
+	invitation, res, err := client.Organizations.InviteUser(context.Background(), *currentModel.OrgId, invitationReq)
+	if err != nil {
+		_, _ = log.Warnf("Create - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+	currentModel.Id = &invitation.ID
+
+	return Read(req, prevModel, currentModel)
```

Comment:
Instead of using READ api call, Can we leverage a common method just creating Model with necessary data.

Reply:
```suggestion
	// Response
	return handler.ProgressEvent{
		OperationStatus: handler.Success,
		ResourceModel:   readAtlasOrgInvitation(invitation, currentModel),
	}, nil


```

Reply:
fixed